packages/flutter/lib/src/material/input_decorator.dart

```diff
@@ -4314,3 +4311,335 @@ class InputDecorationTheme with Diagnosticable {
     properties.add(DiagnosticsProperty<BoxConstraints>('constraints', constraints, defaultValue: defaultTheme.constraints));
   }
 }
+
+class _InputDecoratorDefaultsM2 extends InputDecorationTheme {
+  const _InputDecoratorDefaultsM2(this.context)
+      : super();
+
+  final BuildContext context;
+
+  @override
+  TextStyle? get inputStyle => Theme.of(context).textTheme.subtitle1;
+
+  @override
+  TextStyle? get hintStyle => MaterialStateTextStyle.resolveWith((Set<MaterialState> states) {
```

Comment:
In a future PR, you could optimize the cases where a closure isn't need (like all of them?):
```dart
TextStyle? getHintStyle((Set<MaterialState> states) {
  if (states.contains(MaterialState.disabled)) {
    return TextStyle(color: Theme.of(context).disabledColor);
  }
  return TextStyle(color: Theme.of(context).hintColor);
}
TextStyle? get hintStyle => MaterialStateTextStyle.resolveWith(getHintStyle);